assertj-core/src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -1288,42 +1288,6 @@ public AbstractListAssert<?, List<? extends Tuple>, Tuple, ObjectAssert<Tuple>>
     return newListAssertInstanceForMethodsChangingElementType(values).as(description);
   }
 
-  /**
-   * Extract the values from Iterable's elements under test by applying an extracting function on them. The returned
-   * iterable becomes the instance under test.
-   * <p>
-   * It allows to test values from the elements more safely than by using {@link #extracting(String)}, as it
-   * doesn't utilize introspection.
-   * <p>
-   * Let's have a look at an example:
-   * <pre><code class='java'> // Build a list of TolkienCharacter, a TolkienCharacter has a name, and age and a Race (a specific class)
-   * // they can be public field or properties, both can be extracted.
-   * List&lt;TolkienCharacter&gt; fellowshipOfTheRing = new ArrayList&lt;TolkienCharacter&gt;();
-   *
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Frodo&quot;, 33, HOBBIT));
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Sam&quot;, 38, HOBBIT));
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Gandalf&quot;, 2020, MAIA));
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Legolas&quot;, 1000, ELF));
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Pippin&quot;, 28, HOBBIT));
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Gimli&quot;, 139, DWARF));
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Aragorn&quot;, 87, MAN);
-   * fellowshipOfTheRing.add(new TolkienCharacter(&quot;Boromir&quot;, 37, MAN));
-   *
-   * // fellowship has hobbitses, right, my presioussss?
-   * assertThat(fellowshipOfTheRing).extracting(TolkienCharacter::getRace).contains(HOBBIT);</code></pre>
-   *
-   * Note that the order of extracted property/field values is consistent with the iteration order of the Iterable under
-   * test, for example if it's a {@link HashSet}, you won't be able to make any assumptions on the extracted values order.
-   *
-   * @param <V> the type of elements extracted.
-   * @param extractor the object transforming input object to desired one
-   * @return a new assertion object whose object under test is the list of values extracted
-   */
-  @CheckReturnValue
-  public <V> AbstractListAssert<?, List<? extends V>, V, ObjectAssert<V>> extracting(Function<? super ELEMENT, V> extractor) {
```

Comment:
As discussed today with @filiphr, removing the overload accepting `Function` will break compatibility with all the existing `Function` implementations that cannot be easily transformed into a `ThrowingExtractor`, such as those coming from third parties.

A better strategy is likely to keep `extracting(Function)` and add to `ThrowingExtractor` factory methods that ease wrapping code that throws exceptions, similar to [how the Spring Framework does](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/function/ThrowingFunction.html#of(org.springframework.util.function.ThrowingFunction)).

I'll analyze the topic again and come up with a better proposal.